src/entity/spatial/Geometry.cpp

```diff
@@ -63,7 +63,7 @@ SpatialReference::Ptr Geometry::getCS() const
 
 void Geometry::apply(Filter& filter)
 {
-    geom::Geometry* geometry = this->geometry();
+    geom::Geometry* geometry = static_cast<const Geometry*>(this)->getGeometry();   // best match call to the protected getter. If this is non-const you can only call the public getter with a const return value.
```

Comment:
This will be solved if you fix the const modifiers of the methods. See the other comment. ;-)

Reply:
See comment above.
Its solved in dcadddeb596ea4d3c0ae96020c2bbeefad3de695